man/NetworkManager.conf.xml

```diff
@@ -306,15 +306,13 @@ no-auto-default=*
       </varlistentry>
       <varlistentry>
         <term><varname>dns</varname></term>
-        <listitem><para>Set the DNS (<filename>resolv.conf</filename>) processing mode.
-        If the key is unspecified, <literal>default</literal> is used,
-        unless <filename>/etc/resolv.conf</filename> is a symlink to
-        <filename>/run/systemd/resolve/stub-resolv.conf</filename>,
-        <filename>/run/systemd/resolve/resolv.conf</filename>,
-        <filename>/lib/systemd/resolv.conf</filename> or
-        <filename>/usr/lib/systemd/resolv.conf</filename>.
-        In that case, <literal>systemd-resolved</literal> is chosen automatically.
-        </para>
+        <listitem><para>Configures the processing mode of DNS information known to
+        NetworkManager. Takes a value in
+        <arg choice='plain'><replaceable>mode</replaceable></arg><arg>,systemd-resolved</arg>
+        form. The primary <replaceable>mode</replaceable> influences how is the system's
+        <filename>resolv.conf</filename> managed, while the supplementary systemd-resolved
+        mode allows passing the DNS configuration to systemd-resolved even if it's not
+        configured as a primary resolver. The following modes are available:</para>
```

Comment:
it seems not optimal to configure it this way. the only setting that makes sense, is either having
```
dns=$ONE_VALUE
```
or
```
dns=$ONE_VALUE_NOT_SDRD,systemd-resolved
```
meaning: you either select one mode; or you select a mode != `systemd-resolved` and followed by `systemd-resolved`.

Wouldn't it be clearer to add separate option like:
```
dns-internal=systemd-resolved
```
?

Reply:
I don't know. Reusing `dns` for both seems more future-proof to me.

There doesn't seem to be a reason (apart from current implementation limitations) why you wouldn't keep an instances of dnsmasq, sd-resolved and perhaps some other dns forwarders all configured at the same time while perhaps defaulting to configuring the system resolver not to use them.

Reply:
I don't see so useful to allow multiple DNS plugins at the same time in
an arbitrary way. The only special exception should be systemd-resolved,
which is the only one that can be used to resolve names for connectivity
check. So, I think we should use it by default for connectivity check
(along with the one specified by the dns= key to be used system-wide).

If we want to allow disabling sd-resolved for connectivity check, we
could add a new key (like Thomas said) as 'dns-internal', or
'concheck-dns' (but perhaps this is too specific and in the future we'll
use it also for something else). The allowed values would be 'none' and
'systemd-resolved', with the latter being the default.

Reply:
Okay. I'm going with systemd-resolved=yes/no, because the setting is not specific to connectivity checking.